lib/util.js

```diff
@@ -738,7 +738,9 @@ function formatSet(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
   value.forEach(function(v) {
     var nextRecurseTimes = recurseTimes === null ? null : recurseTimes - 1;
-    var str = formatValue(ctx, v, nextRecurseTimes);
+    var str = ctx.seen.includes(v) ?
+      ctx.stylize('[Circular]', 'special') :
+      formatValue(ctx, v, nextRecurseTimes);
```

Comment:
@addaleax That would be a semver-major change as the `ctx` is passed to custom inspection functions.